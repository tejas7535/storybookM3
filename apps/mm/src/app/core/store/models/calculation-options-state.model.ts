import { PreflightData } from '@mm/core/services/preflght-data-parser/preflight-data.interface';
import { ThermalCalculationOptionsFormData } from '@mm/steps/calculation-options-step/calculation-selection-step.interface';

export interface CalculationOptionsState {
  options: PreflightData;
  thermalOptions: ThermalCalculationOptionsFormData;
  calculationPerformed: boolean;
  toleranceClasses?: string[];
}
