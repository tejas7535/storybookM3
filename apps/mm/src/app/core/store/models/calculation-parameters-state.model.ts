export interface CalculationParametersState {
  parameters?: CalculationParameters;
}

export interface CalculationParameters {
  RSY_BEARING_TYPE: number;
  RSY_BEARING_SERIES: string;
  IDCO_DESIGNATION: string;
  IDMM_BEARING_SEAT: string;
  IDMM_MEASSURING_METHOD: string;
  IDMM_MOUNTING_METHOD: string;
  IDMM_HYDRAULIC_NUT_TYPE: string;
  IDMM_NUMBER_OF_PREVIOUS_MOUNTINGS: string;
  IDMM_CLEARANCE_REDUCTION_INPUT: string;
  IDMM_INNER_RING_EXPANSION: number;
  IDMM_RADIAL_CLEARANCE_REDUCTION: number;
  IDMM_INNER_SHAFT_DIAMETER: number;
  IDMM_SHAFT_MATERIAL: string;
  IDMM_MODULUS_OF_ELASTICITY: number;
  IDMM_POISSON_RATIO: number;
}
