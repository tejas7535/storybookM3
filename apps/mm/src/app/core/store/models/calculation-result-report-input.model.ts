export interface CalculationResultReportInput {
  hasNestedStructure: boolean;
  meaningfulRound?: boolean;
  title?: string;
  designation?: string;
  abbreviation?: string;
  unit?: string;
  value?: string;
  subItems?: CalculationResultReportInput[];
  titleID?: string;
}
