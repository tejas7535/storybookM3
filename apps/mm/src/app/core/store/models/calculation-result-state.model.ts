import { CalculationResultReportInput } from './calculation-result-report-input.model';

export interface CalculationResultState {
  isLoading: boolean;
  result?: CalculationResult;
  error?: string;
  versions?: { [key: string]: string };
}

export interface CalculationResult {
  inputs: CalculationResultReportInput[];
  startPositions: ResultItem[];
  endPositions: ResultItem[];
  mountingTools: MountingTools;
  mountingRecommendations: string[];
  radialClearance: ResultItemWithTitle[];
  clearanceClasses: ResultItemWithTitle[];
  reportMessages: ReportMessages;
}

export interface ResultItem {
  value: string;
  unit: string;
  abbreviation?: string;
  designation: string;
}

export interface ResultItemWithTitle {
  title: string;
  items: ResultItem[];
}

export interface PumpItem {
  field: string;
  value: string;
  isRecommended: boolean;
}

export interface MountingTools {
  additionalTools: ResultItem[];
  hydraulicNut: ResultItem[];
  pumps: {
    title: string;
    items: PumpItem[];
  };
  locknut: ResultItem[];
  sleeveConnectors: ResultItem[];
}

export interface ReportMessages {
  warnings: string[];
  errors: string[];
  notes: string[];
}

export type ResultType =
  | 'reportInputs'
  | 'startPosition'
  | 'endPosition'
  | 'mountingToolsAndUtilities'
  | 'mountingInstructions'
  | 'reportMessages';

export interface ResultTypeConfig {
  name: ResultType;
}
