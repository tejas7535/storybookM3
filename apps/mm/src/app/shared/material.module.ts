import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MatExpansionModule } from '@angular/material/expansion';
import { MatIconModule } from '@angular/material/icon';
import { MatLegacyButtonModule as MatButtonModule } from '@angular/material/legacy-button';
import { MatLegacyCardModule as MatCardModule } from '@angular/material/legacy-card';
import { MatLegacyDialogModule as MatDialogModule } from '@angular/material/legacy-dialog';
import { MatLegacyFormFieldModule as MatFormFieldModule } from '@angular/material/legacy-form-field';
import { MatLegacyInputModule as MatInputModule } from '@angular/material/legacy-input';
import { MatLegacySelectModule as MatSelectModule } from '@angular/material/legacy-select';
import { MatLegacySlideToggleModule as MatSlideToggleModule } from '@angular/material/legacy-slide-toggle';
import { MatSidenavModule } from '@angular/material/sidenav';
import { MatStepperModule } from '@angular/material/stepper';

@NgModule({
  imports: [
    CommonModule,
    MatCardModule,
    MatSelectModule,
    MatButtonModule,
    MatFormFieldModule,
    MatInputModule,
    MatStepperModule,
    MatSidenavModule,
    MatExpansionModule,
    MatIconModule,
    MatDialogModule,
    MatSlideToggleModule,
  ],
  exports: [
    CommonModule,
    MatCardModule,
    MatSelectModule,
    MatButtonModule,
    MatFormFieldModule,
    MatInputModule,
    MatStepperModule,
    MatSidenavModule,
    MatExpansionModule,
    MatIconModule,
    MatDialogModule,
    MatSlideToggleModule,
  ],
})
export class MaterialModule {}
