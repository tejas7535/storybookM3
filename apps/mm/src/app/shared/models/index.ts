export * from './app-delivery.model';
export * from './bearing-search.model';
export * from './bearings-seats-response.model';
export * from './calculation-request.model';
export * from './maintenance-message';
export * from './peflight-request.model';
export * from './preflight-response.model';
export * from './shaft-material-response.model';
export * from './simple-list-response.model';
