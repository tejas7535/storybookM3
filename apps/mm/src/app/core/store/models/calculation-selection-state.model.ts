import { BearingOption } from '@mm/shared/models';
import { ListValue } from '@mm/shared/models/list-value.model';

export interface CalculationSelectionState {
  bearingResultList: BearingOption[];
  bearing?: Bearing;
  bearingSeats?: StepSelectionValue;
  measurementMethods?: StepSelectionValue;
  mountingMethods?: StepSelectionValue;
  stepper: {
    currentStep: number;
  };
  loading?: boolean;
}

export interface Bearing {
  bearingId: string;
  title: string;
}

export interface StepSelectionValue {
  values: ListValue[];
  selectedValueId?: string;
}
