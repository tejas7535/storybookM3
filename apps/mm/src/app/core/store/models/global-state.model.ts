import { AppDelivery } from '@mm/shared/models';

export interface GlobalState {
  isStandalone: boolean;
  appDelivery: `${AppDelivery}`;
  initialized: boolean;
  isInternalUser: boolean;
}
