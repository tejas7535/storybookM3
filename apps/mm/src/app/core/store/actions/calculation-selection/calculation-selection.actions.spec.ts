import { BearingOption } from '@mm/shared/models';
import { ListValue } from '@mm/shared/models/list-value.model';

import { CalculationSelectionActions } from '.';

describe('CalculationSelectionActions', () => {
  it('should create an action to search bearing list', () => {
    expect(
      CalculationSelectionActions.searchBearingList({ query: '220' })
    ).toMatchSnapshot();
  });

  it('should create searchBearingSuccess action', () => {
    const resultList: BearingOption[] = [{ id: '1', title: 'Bearing 1' }];
    const action = CalculationSelectionActions.searchBearingSuccess({
      resultList,
    });
    expect(action).toMatchSnapshot();
  });

  it('should create resetBearing action', () => {
    const action = CalculationSelectionActions.resetBearing();
    expect(action).toMatchSnapshot();
  });

  it('should create fetchBearingData action', () => {
    const bearingId = '123';
    const action = CalculationSelectionActions.fetchBearingData({ bearingId });
    expect(action).toMatchSnapshot();
  });

  it('should create setBearing action', () => {
    const bearingId = '123';
    const title = 'Bearing Title';
    const action = CalculationSelectionActions.setBearing({ bearingId, title });
    expect(action).toMatchSnapshot();
  });

  it('should create setCurrentStep action', () => {
    const step = 1;
    const action = CalculationSelectionActions.setCurrentStep({ step });
    expect(action).toMatchSnapshot();
  });

  it('should create setBearingSeats action', () => {
    const bearingSeats: ListValue[] = [{ id: '1', text: 'Seat 1' }];
    const action = CalculationSelectionActions.setBearingSeats({
      bearingSeats,
    });
    expect(action).toMatchSnapshot();
  });

  it('should create setBearingSeat action', () => {
    const bearingSeatId = 'seat123';
    const action = CalculationSelectionActions.setBearingSeat({
      bearingSeatId,
    });
    expect(action).toMatchSnapshot();
  });

  it('should create fetchMeasurementMethods action', () => {
    const action = CalculationSelectionActions.fetchMeasurementMethods();
    expect(action).toMatchSnapshot();
  });

  it('should create setMeasurementMethods action', () => {
    const measurementMethods: ListValue[] = [{ id: '1', text: 'Method 1' }];
    const action = CalculationSelectionActions.setMeasurementMethods({
      measurementMethods,
    });
    expect(action).toMatchSnapshot();
  });

  it('should create setMeasurementMethod action', () => {
    const measurementMethod = 'method123';
    const action = CalculationSelectionActions.setMeasurementMethod({
      measurementMethod,
    });
    expect(action).toMatchSnapshot();
  });

  it('should create fetchMountingMethods action', () => {
    const action = CalculationSelectionActions.fetchMountingMethods();
    expect(action).toMatchSnapshot();
  });

  it('should create setMountingMethods action', () => {
    const mountingMethods: ListValue[] = [{ id: '1', text: 'Method 1' }];
    const action = CalculationSelectionActions.setMountingMethods({
      mountingMethods,
    });
    expect(action).toMatchSnapshot();
  });

  it('should create updateMountingMethodAndCurrentStep action', () => {
    const mountingMethod = 'method123';
    const action =
      CalculationSelectionActions.updateMountingMethodAndCurrentStep({
        mountingMethod,
      });
    expect(action).toMatchSnapshot();
  });

  it('should create setMountingMethod action', () => {
    const mountingMethod = 'method123';
    const action = CalculationSelectionActions.setMountingMethod({
      mountingMethod,
    });
    expect(action).toMatchSnapshot();
  });

  it('should create fetchPreflightOptions action', () => {
    const action = CalculationSelectionActions.fetchPreflightOptions();
    expect(action).toMatchSnapshot();
  });

  it('should create setPreflightOptions action', () => {
    const action = CalculationSelectionActions.setPreflightOptions();
    expect(action).toMatchSnapshot();
  });
});
