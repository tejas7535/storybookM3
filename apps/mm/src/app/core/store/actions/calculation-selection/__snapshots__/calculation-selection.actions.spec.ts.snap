// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculationSelectionActions should create an action to search bearing list 1`] = `
Object {
  "query": "220",
  "type": "[CalculationSelection] Search Bearing list",
}
`;

exports[`CalculationSelectionActions should create fetchBearingData action 1`] = `
Object {
  "bearingId": "123",
  "type": "[CalculationSelection] Fetch Bearing data",
}
`;

exports[`CalculationSelectionActions should create fetchMeasurementMethods action 1`] = `
Object {
  "type": "[CalculationSelection] Fetch measurement methods",
}
`;

exports[`CalculationSelectionActions should create fetchMountingMethods action 1`] = `
Object {
  "type": "[CalculationSelection] Fetch mounting methods",
}
`;

exports[`CalculationSelectionActions should create fetchPreflightOptions action 1`] = `
Object {
  "type": "[CalculationSelection] Fetch preflight options",
}
`;

exports[`CalculationSelectionActions should create resetBearing action 1`] = `
Object {
  "type": "[CalculationSelection] Reset Bearing",
}
`;

exports[`CalculationSelectionActions should create searchBearingSuccess action 1`] = `
Object {
  "resultList": Array [
    Object {
      "id": "1",
      "title": "Bearing 1",
    },
  ],
  "type": "[CalculationSelection] Search Bearing Success",
}
`;

exports[`CalculationSelectionActions should create setBearing action 1`] = `
Object {
  "bearingId": "123",
  "title": "Bearing Title",
  "type": "[CalculationSelection] Set Bearing",
}
`;

exports[`CalculationSelectionActions should create setBearingSeat action 1`] = `
Object {
  "bearingSeatId": "seat123",
  "type": "[CalculationSelection] Set Bearing Seat",
}
`;

exports[`CalculationSelectionActions should create setBearingSeats action 1`] = `
Object {
  "bearingSeats": Array [
    Object {
      "id": "1",
      "text": "Seat 1",
    },
  ],
  "type": "[CalculationSelection] Set bearing seats",
}
`;

exports[`CalculationSelectionActions should create setCurrentStep action 1`] = `
Object {
  "step": 1,
  "type": "[CalculationSelection] Set Current Step",
}
`;

exports[`CalculationSelectionActions should create setMeasurementMethod action 1`] = `
Object {
  "measurementMethod": "method123",
  "type": "[CalculationSelection] Set Measurement Method",
}
`;

exports[`CalculationSelectionActions should create setMeasurementMethods action 1`] = `
Object {
  "measurementMethods": Array [
    Object {
      "id": "1",
      "text": "Method 1",
    },
  ],
  "type": "[CalculationSelection] Set measurement methods",
}
`;

exports[`CalculationSelectionActions should create setMountingMethod action 1`] = `
Object {
  "mountingMethod": "method123",
  "type": "[CalculationSelection] Set Mounting Method",
}
`;

exports[`CalculationSelectionActions should create setMountingMethods action 1`] = `
Object {
  "mountingMethods": Array [
    Object {
      "id": "1",
      "text": "Method 1",
    },
  ],
  "type": "[CalculationSelection] Set mounting methods",
}
`;

exports[`CalculationSelectionActions should create setPreflightOptions action 1`] = `
Object {
  "type": "[CalculationSelection] set preflight options",
}
`;

exports[`CalculationSelectionActions should create updateMountingMethodAndCurrentStep action 1`] = `
Object {
  "mountingMethod": "method123",
  "type": "[CalculationSelection] Update Mounting Method and Current Step",
}
`;
